### DISCUSSION TITLE
New Writer API for writing VTT?

### DISCUSSION THREAD

**Discussion by mattpetters** [openai/whisper#880] [Q&A] [answered]
Hi!

I am trying to use this notebook (https://github.com/dmarx/video-killed-the-radio-star) which leverages Whisper to transcribe audio for then generating a music video.

That notebook hasn't been updated in a while and it looks like it now fails at the lines regarding writing VTT:

```
    #######################
    # save transcriptions #
    #######################

    transcriptions = {}
    transcription_root = root / 'whispers'
    transcription_root.mkdir(parents=True, exist_ok=True)
    for k in whispers:
        outpath = str( transcription_root / f"{k}.vtt" )
        transcriptions[k] = outpath
        with open(outpath,'w') as f:
            # to do: upstream PR to control verbosity
           # FAILS HERE, write_vtt not a function
            whisper.utils.write_vtt(
                whispers[k]["segments"], # ...really?
                file=f
            ) 
    storyboard.params.whisper.transcriptions = transcriptions

```

Can someone help me understand the way to fix this code? I saw some example showing something along these lines:

```
writer = writer.utils.get_writer(output_type, output_path)

```

Thanks for your time!

Matt

**Comment by jongwook** [comment, marked as answer]
Hi, I made a breaking change recently. If you install an older version via PIP:

```python
pip uninstall whisper
pip install openai-whisper==20230117
```

The notebook should work.

**Reply by mattpetters** [reply]
Awesome, thanks a ton. I will try that. Out of curiosity, can you give a simple example of the new version/usage?

**Reply by jongwook** [reply]
It could be something like:

```python
from whisper.utils import get_writer

writer = get_writer("vtt", str(transcription_root))
writer(whispers[k], f"{audio_fpath}.{k}")
```

I added `"{k}"` in the filename because the notebook was running transcription on `tiny` and then `large`; up to you to change the file name though!

**Reply by mattpetters** [reply]
Thanks so much!

**Reply by beldabeast67** [reply]
Hi!  I still can't get this to work by installing an older PIP version. I'm not a coder so I'm a little confused on what other steps to take.

**Reply by mattpetters** [reply]
@beldabeast67 take a look at this issue over here: https://github.com/dmarx/video-killed-the-radio-star/issues/101

**Reply by beldabeast67** [reply]
@mattpetters Thank you so much!  That made it much simpler for my feeble musician brain.  It's working now.
